File: input_processing_output/python/retirement_calculator.py
# Pratice 6. Retirement Calculator
# Output:
#   What is your current age? 27
#   At what age would you like to retire? 70
#   You have 43 years left until you can retire.
#   It's 2016, so you can retire in 2059.
# Constraint:
#   - Agian, be sure to convert the input to numerical data
#     before doing any math.
#   - Don't hard-code the current year into your program.
#     Get it from the system time via your programming language.

#!/usr/bin/env python

import sys
from datetime import datetime

def input_process(in_question):
    return input(in_question) if sys.version_info >= (3,0) else raw_input(in_question)


if __name__ == '__main__':
    current_age_str = input_process('What is your current age? ')
    retire_age_str = input_process('At what age would you like to retire? ')

    try:
        current_age = int(current_age_str)
        retire_age = int(retire_age_str)
    except:
        print ('You must input only numbers.')
    else:
        if current_age >= retire_age:
            print('You must input retire age lagger than current age.')
            exit(1)

        left_year = retire_age - current_age
        current_year = datetime.today().year

        print_str = 'You have %s years left until you can retire.' % (left_year)
        print_str += '\n'
        print_str += 'It\'s %s, so you can retire in %s.' % (current_year, current_year + left_year)

        print (print_str)

File: input_processing_output/go/retirement_calculator.go
/*
 * Pratice 6. Retirement Calculator
 * Output:
 *   What is your current age? 27
 *   At what age would you like to retire? 70
 *   You have 43 years left until you can retire.
 *   It's 2016, so you can retire in 2059.
 * Constraint:
 *   - Agian, be sure to convert the input to numerical data
 *     before doing any math.
 *   - Don't hard-code the current year into your program.
 *     Get it from the system time via your programming language.
 */

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
	"time"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Println("What is your current age? ")
	current_age_str, _ := reader.ReadString('\n')
	fmt.Println("At what age would you like to retire? ")
	retire_age_str, _ := reader.ReadString('\n')

	current_age, _ := strconv.Atoi(strings.TrimSpace(current_age_str))
	retire_age, _ := strconv.Atoi(strings.TrimSpace(retire_age_str))

	if current_age >= retire_age {
		fmt.Println("You must input retire age lagger than current age.")
		return
	}

	left_year := retire_age - current_age

	t := time.Now()
	current_year := t.Year()

	print_str := fmt.Sprintf("You have %d years left until you can retire.", left_year)
	print_str += "\n"
	print_str += fmt.Sprintf("It's %d, so you can retire in %d.", current_year, current_year+left_year)

	fmt.Println(print_str)
}
